app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -408,10 +431,6 @@ export class CreateBranch extends React.Component<
       }
     }
 
-    if (this.state.branchName !== branchName) {
```

Comment:
Shouldn't we keep this check after `await checkRepoRules`?

The chance for something bad to happen is slow, but I tend to be paranoid around async stuff 😂

Reply:
That check is here to be after the network call of `await api.fetchRepoRulesForBranch` of which it is directly after to short circuit anymore work if it has changed in the time it took to retrieve the repo rules. 

I will go ahead and put another check in since I did introduce another async method. Tho,  `await parseRepoRules`,  is only async due to the `getBooleanConfigValue` so it is not a network call, thus, I believe the risk is much smaller but, technically still there is a chance for a blip for users with commit signing repo rules.